Tests/Marketplace/marketplace_services.py

```diff
@@ -3007,7 +3026,212 @@ def copy_author_image(self, production_bucket, build_bucket, images_data, storag
+                url_path = Path(path)
+                image_name = url_path.name
+
+                image_gcp_path = Path(gcs_pack_path, BucketUploadFlow.README_IMAGES, image_name)
+                url_to_replace_linking_to_dist = os.path.join(google_api_readme_images_url,
+                                                              BucketUploadFlow.README_IMAGES, image_name)
+
+                line = line.replace(url, str(url_to_replace_linking_to_dist))
+
+                urls_list.append({
+                    'original_read_me_url': url,
+                    'new_gcs_image_path': image_gcp_path,
+                    'image_name': image_name
+                })
+
+            print(line, end='')
+
+        return urls_list
+
+    @staticmethod
+    def download_readme_image_from_url_and_upload_to_gcs(readme_original_url: str, gcs_storage_path: str,
+                                                         image_name: str, storage_bucket):
+        """
+        Download the image from the endpoint url and save locally.
+        Upload The image to gcs.
+        Remove the Temp file.
+
+        Args:
+             readme_original_url (str): The original url that was in the readme file
+             gcs_storage_path (str): The path to save the image on gcp (was calculated in collect_images_from_readme_
+             and_replace_with_storage_path)
+             image_name(str): The name of the image we want to save
+             storage_bucket (google.cloud.storage.bucket.Bucket): gcs bucket where images will be uploaded.
+
+        """
+        # Open the url image, set stream to True, this will return the stream content.
+        readme_original_url = urllib.parse.urlparse(readme_original_url)
+        r = requests.get(readme_original_url.geturl(), stream=True)
+
+        # Check if the image was retrieved successfully
+        if r.status_code == 200:
+            # Set decode_content value to True, otherwise the downloaded image file's size will be zero.
+            r.raw.decode_content = True
+
+            with open(image_name, 'wb') as f:
+                shutil.copyfileobj(r.raw, f)
+
+            # init the blob with the correct path to save the image on gcs
+            readme_image = storage_bucket.blob(gcs_storage_path)
+
+            # load the file from local memo to the gcs
+            with open(image_name, "rb") as image_file:
+                readme_image.upload_from_file(image_file)
+                logging.info('Uploaded the image to gcs')
+
+            # remove local saved image
+            os.remove(image_name)
+
+            logging.info(f'Image sucessfully Downloaded: {image_name}')
+            return True
+        else:
+            logging.error(f'Image {image_name} could not be retreived status code {r.status_code}')
+            return False
+
+    def upload_images(self, index_folder_path, storage_bucket, storage_base_path, diff_files_list, override_all_packs,
+                      marketplace=''):
```

Comment:
Let's give `xsoar` as a default value for marketplace.

Reply:
Done